internal/cli/cmd/sysdump.go

```diff
@@ -29,6 +29,7 @@ func newCmdSysdump() *cobra.Command {
 		Short: "Collects information required to troubleshoot issues with Cilium and Hubble",
 		Long:  ``,
 		RunE: func(cmd *cobra.Command, args []string) error {
+			sysdumpOptions.CiliumNamespace = namespace
```

Comment:
> If both flags are set, then the `--namespace` global flag comes with priority.

I don't think this change works as expected. When specifying only `--cilium-namespace`, the given namespace is no longer used but instead the default for `--namespace` (i.e. `kube-system`) is used.

Reply:
You are right! 

How about following an order like:

1. if `-cilium-namespace` is provided then use it
2. if `-cilium-namespace` is not provided, then use the `--namespace` (if provided) instead 
3. if both are provided `-cilium-namespace` comes with priority
4. if none is provided then we fall back to the default of `--namespace` which is `kube-system`.

Would that work?

Reply:
Sounds good to me, thanks @ChrsMark. We should probably also document this order of precedence in these flags' help messages.